lms/static/scripts/frontend_apps/index.js

```diff
@@ -15,12 +15,10 @@ startRpcServer();
 const rootEl = document.querySelector('#app');
 const config = JSON.parse(document.querySelector('.js-config').textContent);
 
-const mode = config.mode || 'content-item-selection';
```

Comment:
There's no need for JavaScript to have a fallback default for the mode -- the Python always puts the mode in the JSConfig

Reply:
:+1: This is the kind of stuff that makes it very hard to reason about when debugging. You send through `None` expecting things to break and it carries on but behaves differently.